dts/bindings/display/galaxycore,gc9x01x.yaml

```diff
@@ -0,0 +1,181 @@
+# Copyright (c) 2023 Mr Beam Lasers GmbH.
+# Copyright (c) 2023 Amrith Venkat Kesavamoorthi <amrith@mr-beam.org>
+# Copyright (c) 2023 Martin Kiepfer <mrmarteng@teleschirm.org>
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  GC9X01X display driver.
+
+  This driver implements support for various GC9xxx graphics
+  controllers and different display sizes. It has been validated
+  for following controllers:
+   - GC91A01A: (Waveshare 240x240, 1.28inch round lcd display 240x240)
+
+  Here is an example to define a display interface:
+
+  / {
+    pwmleds {
+      compatible = "pwm-leds";
+      backlight_pwm: backlight_pwm {
+        pwms = <&ledc0 0 PWM_MSEC(10) PWM_POLARITY_NORMAL>;
+      };
+    };
+  };
+
+  &spi2 {
+    gc9a01a_lcd: gc9a01a_lcd@0 {
+      compatible = "galaxycore,gc9x01x";
+      reg = <0>;
+      spi-max-frequency = <100000000>;
+      cmd-data-gpios = <&gpio0 4 GPIO_ACTIVE_HIGH>;
+      reset-gpios = <&gpio0 8 GPIO_ACTIVE_LOW>;
+      backlight-pwm = <&backlight_pwm>;
+
+      width = <240>;
+      height = <240>;
+      pixel-format = <GC9X01X_PIXEL_FORMAT_RGB565>;
+    };
+  };
+
+
+compatible: "galaxycore,gc9x01x"
+
+include: [spi-device.yaml, display-controller.yaml]
+
+properties:
+  reset-gpios:
+    type: phandle-array
+    required: true
+    description: RESET pin
+
+      The RESET pin of the GC9X01X is active low.
+      If connected directly the MCU pin should be configured
+      as active low.
+
+  cmd-data-gpios:
+    type: phandle-array
+    required: true
+    description: Command/Data pin
+      The Data/Command pin of the GC9X01X is to be configured
+      high(1) for data, low(0) for command.
+
+  pixel-format:
+    type: int
```

Comment:
While taking a deeper look I found that there is a common (?) definition in `display-controller.yaml`. Isn't it better to reuse this?
It uses int type as well.

Reply:
TBH, I did not get it. You mean to add pixel format to display-controller.yaml?

Reply:
While I cannot say if this would be the right thing to do, I guess `rotation` and `inversion` might be in the same boat if we'd consider moving `pixel-format` to the common layer?

Reply:
There already is a public header file and a yaml definition for the pixel-format configuration (`include/dt-bindings/display/panel.h` and `display-controller.yaml`). I would simply use that. This also makes the internal conversion obsolete, as these definitions are already in line with the lcd driver api.

Reply:
I see, thanks for the point, but to me include/dt-bindings/display/panel.h looks a bit weird. Not sure why string enum could not be used. But that can be improved/fixed later. Btw, pixel-format should only be used to describe hardware, also "display-controller <-> panel-connection".

Reply:
I used the existing pixel format definition from `lcd-controller.yaml` and `panel.h` for now. 
For orientation I used your approach with string enums, which could work commonly for all display drivers offering rotation/orientation feature.

For now the mixed usage of existing defines (pixel-format from panel.h) and the string enums for orientation is working fine with the benefit, that there is not conversion of configuration needed. The configured values match the display api defines.

I get your point but I think there needs to be a separate PR to clean up these common display parameters.